### PR TITLE
[Security] remove deprecated features

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Security/Http/Logout/LogoutUrlGenerator.php [hunk 1]
```diff
@@ -48,21 +48,8 @@ public function __construct(RequestStack $requestStack = null, UrlGeneratorInter
      * @param CsrfTokenManagerInterface|null $csrfTokenManager A CsrfTokenManagerInterface instance
      * @param string|null                    $context          The listener context
      */
-    public function registerListener($key, $logoutPath, $csrfTokenId, $csrfParameter, CsrfTokenManagerInterface $csrfTokenManager = null/*, string $context = null*/)
+    public function registerListener($key, $logoutPath, $csrfTokenId, $csrfParameter, CsrfTokenManagerInterface $csrfTokenManager = null, $context = null)
```
Comment (chalasr) [inline src/Symfony/Component/Security/Http/Logout/LogoutUrlGenerator.php]: should be `string $context`

File: src/Symfony/Component/Security/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
 =========
 
+4.0.0
+-----
+
+ * The `AbstractFormLoginAuthenticator::onAuthenticationSuccess()` was removed.
+   You should implement this method yourself in your concrete authenticator.
+ * removed the `AccessDecisionManager::setVoters()` method
+ * removed the `RoleInterface`
+ * added a sixth `$context` argument to the `LogoutUrlGenerator::registerListener()`
```
Comment (ogizanagi) [inline src/Symfony/Component/Security/CHANGELOG.md]: This one is already done in #22740

**Review by nicolas-grekas — APPROVED** [review]
:+1:

**Comment by nicolas-grekas** [comment]
FC layer missing in fact

**Comment by xabbuh** [comment]
see #22830

**Comment by nicolas-grekas** [comment]
rebase needed

**Comment by xabbuh** [comment]
rebased, tests pass

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**